beacon-chain/db/state.go

```diff
@@ -48,6 +50,13 @@ func (db *BeaconDB) InitializeState(genesisTime uint64, deposits []*pb.Deposit)
 			return err
 		}
 
+		for i, validator := range beaconState.ValidatorRegistry {
+			h := hashutil.Hash(validator.Pubkey)
+			if err := validatorBkt.Put(h[:], []byte(strconv.Itoa(i))); err != nil {
```

Comment:
why do you have to convert the int to a string to a byte array? 

Maybe use the `encoding/binary` package? 
https://godoc.org/encoding/binary#PutUvarint

Reply:
By the way, I had to prove this to myself in a bench.

Results:
```
goos: linux
goarch: amd64
BenchmarkStrconv-4              30000000                47.3 ns/op
BenchmarkBinaryEncoding-4       300000000                5.77 ns/op
PASS
ok      _/tmp/bench     3.770s
```

Test:
```go
import (
  "encoding/binary"
  "strconv"
  "testing"
)

func BenchmarkStrconv(b *testing.B) {
  for i := 0; i < b.N; i++ {
    _ = []byte(strconv.Itoa(i))
  }
}

func BenchmarkBinaryEncoding(b *testing.B) {
  for i := uint64(0); i < uint64(b.N); i++ {
    buf := make([]byte, binary.MaxVarintLen64)
    _ = binary.PutUvarint(buf, i)
  }
}
```

Please fix here and everywhere else in the package. :)

Reply:
Excellent point!